docs/hal.md

```diff
@@ -0,0 +1,24 @@
+# Hardware Abstraction Layer (HAL)
+HAL is an abstraction layer for confidential computing. It is an in-memory virtual machine (VM). For Cocos, HAL is used with SEV-SNP to provide a fully encrypted VM.
```

Comment:
Add explanation of the acronym - (Hardware Abstraction Layer)

Also explain that SVMs act as bare-metal machines (isolated from host OS), so we need to put some runtime in them in order to run the applications.

Reply:
I have added a more detailed description of HAL.